modular/blocksyncer/modules/bucket/bucket_handle.go

```diff
@@ -16,25 +18,28 @@ import (
 )
 
 var (
-	EventCreateBucket            = proto.MessageName(&storagetypes.EventCreateBucket{})
-	EventDeleteBucket            = proto.MessageName(&storagetypes.EventDeleteBucket{})
-	EventUpdateBucketInfo        = proto.MessageName(&storagetypes.EventUpdateBucketInfo{})
-	EventDiscontinueBucket       = proto.MessageName(&storagetypes.EventDiscontinueBucket{})
-	EventCompleteMigrationBucket = proto.MessageName(&storagetypes.EventCompleteMigrationBucket{})
+	EventCreateBucket             = proto.MessageName(&storagetypes.EventCreateBucket{})
+	EventDeleteBucket             = proto.MessageName(&storagetypes.EventDeleteBucket{})
+	EventUpdateBucketInfo         = proto.MessageName(&storagetypes.EventUpdateBucketInfo{})
+	EventDiscontinueBucket        = proto.MessageName(&storagetypes.EventDiscontinueBucket{})
+	EventCompleteMigrationBucket  = proto.MessageName(&storagetypes.EventCompleteMigrationBucket{})
+	EventToggleSPAsDelegatedAgent = proto.MessageName(&storagetypes.EventToggleSPAsDelegatedAgent{})
 )
 
 var BucketEvents = map[string]bool{
-	EventCreateBucket:            true,
-	EventDeleteBucket:            true,
-	EventUpdateBucketInfo:        true,
-	EventDiscontinueBucket:       true,
-	EventCompleteMigrationBucket: true,
+	EventCreateBucket:             true,
+	EventDeleteBucket:             true,
+	EventUpdateBucketInfo:         true,
+	EventDiscontinueBucket:        true,
+	EventCompleteMigrationBucket:  true,
+	EventToggleSPAsDelegatedAgent: true,
 }
 
 type OffChainStatus int
 
 const (
 	OffChainStatusIsLimited OffChainStatus = 1 << iota // 1
+	OffChainStatusSpAsDelegatedAgentDisabled
```

Comment:
would be better to write a comment like // 1 << 1 which is 00000010, just so might be better to understand later.

Reply:
fixed